pandas/core/strings.py

```diff
@@ -502,6 +502,85 @@ def f(x):
                                dtype=object)
     return result, name
 
+def str_extractall(arr, pat, flags=0):
+    """
+    Find all non-overlapping groups in each 
+    subject string in the Series using regular expression pat.
+
+    Parameters
+    ----------
+    pat : string
+        Regular expression pattern
+    flags : int, default 0 (no flags)
+        re module flags, e.g. re.IGNORECASE
+
+    Returns
+    -------
+    A Series with one element for each match if there is only
+    one group in pat. Otherwise, a DataFrame with one row 
+    for each match and one column for each group.
+
+    Examples
+    --------
+    A pattern with one group will return a Series.
+    Indices with no matches will not appear in the result.
+
+    >>> S = Series(["a1a2", "b1", "c1"], ["A", "B", "C"])
+    >>> S.str.extractall("[ab](?P<digit>\d)")
+    A    1
+    A    2
+    B    1
+    Name: digit, dtype: object
+    
+    A pattern with multiple groups will return a DataFrame.
+    
+    >>> S.str.extractall("(?P<letter>[ab])(?P<digit>\d)")
+      letter digit
+    A      a     1
```

Comment:
why would this NOT be a multi-index here? Having a duplicated index is not at all convenient.